app/controllers/branding_groups_controller.rb

```diff
@@ -109,6 +134,23 @@ def self.merge_curator_params(curator_list, collection, user)
     users
   end
 
+  # allow mass assignment of studies from collection edit view
+  def self.find_studies_from_accessions(study_list, user)
+    accessions = param_to_array(study_list)
+    # skip checking group shares for performance reasons
+    Study.where(:accession.in => accessions).select { |study| study.can_view?(user, check_groups: false) }
+  end
+
+  # covert a comma- or space-delimited string to an array of strings, removing empty values
+  def self.param_to_array(param)
+    param.split(/[,\s]/).map(&:strip).reject(&:blank?)
+  end
+
+  # detect which studies could not be saved (either don't exist or curator does't have permission)
+  def self.get_missing_studies(original_accessions, studies)
+    original_accessions - studies.map(&:accession)
```

Comment:
Such an elegant construct in Ruby, `array - array`.  I didn't know this was possible.

For reference, JavaScript and Python don't support this (to their discredit, IMO).

Reply:
I find that surprising - I would have thought at least Python would have that.  I use `-` and `&` a lot when doing array "math" (difference and intersection, respectively).